uaclient/files/notices.py

```diff
@@ -0,0 +1,206 @@
+        override_dir: Optional[str] = None,
+    ):
+        self.directory = override_dir
+        if not override_dir:
+            self.override = False
+        else:
+            self.override = True
+        self.is_root = root_mode
+
+    def add(self, notice_details: NoticeFileDetails, description: str):
+        """Adds a notice file.
+
+        Fails to create the file if the user is not root.
+
+        :param notice_details: Holds details concerning the notice file.
+        :param description: The content to be written to the notice file.
+        """
+        if self.is_root:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            if not self.directory:
+                self.directory = (
+                    defaults.NOTICES_PERMANENT_DIRECTORY
+                    if is_notice_permanent
+                    else defaults.NOTICES_TEMPORARY_DIRECTORY
+                )
+            notice_files = self.get_all_notice_files()
+            if notice_files is None:
+                os.makedirs(self.directory)
+                notice_files = []
+            labels_present = [file.split("-")[-1] for file in notice_files]
+            if label in labels_present:
+                return
+            filename = "{}-{}".format(file_order_id, label)
+            system.write_file(
+                os.path.join(self.directory, filename), description
+            )
+        else:
+            event.warning("Trying to add a notice as non-root user")
+            raise exceptions.NonRootUserError
+
+    def remove(self, notice_details: NoticeFileDetails):
+        """Deletes a notice file.
+
+        :param notice_details: Holds details concerning the notice file.
+        :param description: The content to be written to the notice file.
+        :raises: NonRootUserError when called by a user who is not root.
+        """
+        if self.is_root:
+            label = notice_details.label
+            file_order_id = notice_details.order_id
+            is_notice_permanent = notice_details.is_permanent
+            if not self.directory:
+                self.directory = (
+                    defaults.NOTICES_PERMANENT_DIRECTORY
+                    if is_notice_permanent
+                    else defaults.NOTICES_TEMPORARY_DIRECTORY
+                )
+            filename = "{}-{}".format(file_order_id, label)
+            notice_files = self.get_all_notice_files()
+            if notice_files:
+                label_file = [
+                    file for file in notice_files if file == filename
+                ]
+                if len(label_file) == 1:
+                    system.remove_file(
+                        os.path.join(self.directory, label_file[0])
+                    )
+        else:
+            event.warning("Trying to remove notice as non-root user")
+            raise exceptions.NonRootUserError
```

Comment:
I'd like to reconsider this as well. We can probably just not do anything here rather than raise an error. Another thing we can discuss